From 1ef2fe07839ea7c1f09faa56ef2ba573b04bcaea Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:42 -0400
Subject: [cpufreq] p4: don't initialize part of policy set by core

Message-id: <1409235879-1283-69-git-send-email-prarit@redhat.com>
Patchwork-id: 89445
O-Subject: [RHEL7.1 PATCH BZ 1134639 068/185] cpufreq: p4: don't initialize part of policy set by core
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit c8fb6e9a906c97d86c9d7dda20e0141f952db678
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Oct 3 20:28:52 2013 +0530

    cpufreq: p4: don't initialize part of policy set by core

    Many common initializations of struct policy are moved to core now and hence
    this driver doesn't need to do it. This patch removes such code.

    Most recent of those changes is to call ->get() in the core after calling
    ->init().

    Cc: David S. Miller <davem@davemloft.net>
    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/p4-clockmod.c b/drivers/cpufreq/p4-clockmod.c
index 4fe6d4c..6164c1c 100644
--- a/drivers/cpufreq/p4-clockmod.c
+++ b/drivers/cpufreq/p4-clockmod.c
@@ -230,7 +230,6 @@ static int cpufreq_p4_cpu_init(struct cpufreq_policy *policy)
  /* the transition latency is set to be 1 higher than the maximum
   * transition latency of the ondemand governor */
  policy->cpuinfo.transition_latency = 10000001;
- policy->cur = stock_freq;
 
  return cpufreq_table_validate_and_show(policy, &p4clockmod_table[0]);
 }
-- 
1.7.1